### ISSUE TITLE
Apps that don't close when requested cause Lomiri to crash or hang

### ISSUE THREAD

**Issue by UniversalSuperBox** [ubports/ubuntu-touch#1417]
- Device: All
- Channel: rc/devel
- Build: all since merge of https://github.com/ubports/qtmir/pull/29

### Steps to reproduce

1. Produce/find an application which does not handle the mirclient request to close a window (or freeze an application by attaching a debugger or something)
1. Close the application by swiping it away in the Spread

### Expected behavior

The app is closed

### Actual behavior

Either Lomiri crashes, taking the user back to the Spinner, or Lomiri appears to continue to work normally but will not launch any new apps.

### Logfiles and additional information

The app not closing cleanly, and the subsequent use-after-free, can be seen in the log output following the app being closed: https://paste.ubuntu.com/p/hbNzvPKWsh/

**Comment by UniversalSuperBox** [comment]
Closed by https://github.com/ubports/qtmir/pull/51

**Comment by PeterNerlich** [comment]
Is there any known application to test this with?

**Comment by UniversalSuperBox** [comment]
https://open-store.io/app/cubechron.craig
This is the only one

**Comment by Rondarius** [comment]
device: Hammerhead

os version: RC 2020-W16

When I tries to reproduce the bug, it does not crash Lomiri. I am able to launch other apps. But the background stays blurred out for about 3-4 seconds and the icon remains in the launcher for about a total time of 7-8 seconds. During this time I am able to launch other apps.

**Comment by Bolly** [comment]
> device: Hammerhead
> 
> os version: RC 2020-W16
> 
> When I tries to reproduce the bug, it does not crash Lomiri. I am able to launch other apps. But the background stays blurred out for about 3-4 seconds and the icon remains in the launcher for about a total time of 7-8 seconds. During this time I am able to launch other apps.

Exactly the same. I checked this on a BQM10FHD running RC 2020 W-16 with the application cubechron.craig.

**Comment by PeterNerlich** [comment]
BQ Aquaris E4.5
2020-W16

Installed and opened the app, closed it again. Didn't notice anything wrong. Wanted to try again, but app would load indefinitively when started. Tried again after reboot, observed the same behaviour as described in the previous comments. Subsequent app starts worked fine.

**Comment by clusterfreak** [comment]
Device: Turbo
2020-W16

No Problem with the app Cube Chron. (several starts) 
No interactions with other apps

**Comment by j2g2rp** [comment]
I tried about 10 times in Nexus 5 and M10FHD (last RC both).
I just saw the first time on Nexus 5 that the morph launcher on the dash, which I have  pinned, disappeared (there was an empty hole). I tried to reproduce opening cube app and closing it in several modes ( doing things like closing multiple apps simultaneously ) but I couldn't. The app were closed correctly, just the 7 or 8 seconds to disappear the cube icon from the dash that was reported before and the 3 seconds of the blurred background.
